Instrukcja adapter - review

cpprian opened the pull request.

Commit:
change value for a starting point

Commit:
fix: display simulation in application windows instead of separate window

Commit:
change name: MyAdapter -> Job2DriverAdapter

Commit:
fix: correct drawing Figure Joe 1

Commit:
add Figure Joe 2

Commit:
add constants in TestJobs2Patterns

Commit:
fix: clear draw panel

Commit:
Rename Drawer Simulator in app

Commit:
init LineDrawerAdapter

Commit:
refator Job2DriverAdapter

Commit:
impl LineDrawerAdapter

Commit:
add Line Drawer Simulator to Application

kowallus commented on .idea/compiler.xml:
Commits polluted with project files (.idea, etc.).

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/Job2DriverAdapter.java:
```diff
@@ -6,13 +6,15 @@
+ * driver adapter to drawer
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class Job2DriverAdapter implements Job2dDriver {
```
The name is better but it does not indicate the purpose of the adapter or what was actually adapted.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:
```diff
@@ -0,0 +1,38 @@
+public class LineDrawerAdapter implements Job2dDriver {
+    private int startX = 0, startY = 0;
+    private DrawPanelController drawPanelController;
+    private AbstractLine line;
```
ILine interface could be used here

kowallus commented on src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:
```diff
@@ -16,6 +16,13 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
 	public void actionPerformed(ActionEvent e) {
-		FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+		switch (e.getActionCommand()) {
+		case "Figure Joe 1":
```
Menu action should not depend on its caption. Enum or separate listeners would be better. At least you should consider constants.

kowallus commented on src/test/java/edu/kis/powp/jobs2d/TestJobs2dPatterns.java:
```diff
@@ -83,7 +97,7 @@ public static void main(String[] args) {
 				Application app = new Application("2d jobs Visio");
 				DrawerFeature.setupDrawerPlugin(app);
-				setupDefaultDrawerVisibilityManagement(app);
+//				setupDefaultDrawerVisibilityManagement(app);
```
In revision repo it is safe to delete code instead of using a comment.

kowallus reviewed (commented):
4 pts

Specific remarks are inlined in the code.

cpprian committed:
+ delete Eclipse meta-data
+ impl DriverCommnad interface
+ create SetPositionCommand and OperateToCommand classes
+ impl SetPositionCommand

cpprian committed:
full implementation - SetPositionCommand OperateToCommand

cpprian committed:
impl ComplexCommand

cpprian committed:
impl ComplexCommand (TODO: fix Rectangle and Triangle)
